### PR TITLE
Added `PanopticSegmentationGenerator`

### PR DESCRIPTION
**Pull request by srihari-humbarwadi**
# Description
added `PanopticSegmentationGenerator` layer, which is needed for merging instance and semantic segmentation masks.

## Type of change
- [x] A new research paper code implementation

## Tests
  - `python -m official.vision.beta.projects.panoptic_maskrcnn.modeling.layers.panoptic_segmentation_generator_test`

## Checklist

- [x] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [x] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py [hunk 1]
```diff
@@ -112,8 +129,11 @@ def panoptic_maskrcnn_resnetfpn_coco() -> cfg.ExperimentConfig:
           init_checkpoint_modules=['backbone'],
           model=PanopticMaskRCNN(
               num_classes=91, input_size=[1024, 1024, 3],
+              panoptic_segmentation_generator=PanopticSegmentationGenerator(
+                  output_size=[1024, 1024],
+                  stuff_classes_offset=90),
               segmentation_model=SEGMENTATION_MODEL(
-                  num_classes=91,
+                  num_classes=110,
```
Comment (arashwan) [inline official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py:136]: Can you leave comment here?

Comment (arashwan) [reply official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py:136]: about num of classes.

File: official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py [hunk 2]
```diff
@@ -60,6 +75,8 @@ class PanopticMaskRCNN(maskrcnn.MaskRCNN):
   include_mask = True
   shared_backbone: bool = True
   shared_decoder: bool = True
+  panoptic_segmentation_generator: PanopticSegmentationGenerator = \
```
Comment (arashwan) [inline official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py:78]: Can you add an option disable having this branch?

Comment (arashwan) [reply official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py:78]: A flag that can decide whether or not to have this as part of the outputs during eval.

File: official/vision/beta/projects/__init__.py [hunk 3]
```diff
@@ -12,3 +12,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from official.vision.beta.projects import panoptic_maskrcnn
```
Comment (arashwan) [inline official/vision/beta/projects/__init__.py:15]: remove? why do you want that?

File: official/vision/beta/projects/panoptic_maskrcnn/__init__.py [hunk 4]
```diff
@@ -25,3 +25,6 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+from official.vision.beta.projects.panoptic_maskrcnn import configs
```
Comment (arashwan) [inline official/vision/beta/projects/panoptic_maskrcnn/__init__.py:29]: again why?

File: official/vision/beta/projects/panoptic_maskrcnn/configs/__init__.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Configs package definition."""
+
+from official.vision.beta.projects.panoptic_maskrcnn.configs import panoptic_maskrcnn
```
Comment (arashwan) [inline official/vision/beta/projects/panoptic_maskrcnn/configs/__init__.py:18]: same.

File: official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py [hunk 6]
```diff
@@ -72,6 +73,18 @@ class DataConfig(maskrcnn.DataConfig):
   parser: Parser = Parser()
 
 
+@dataclasses.dataclass
+class PanopticSegmentationGenerator(hyperparams.Config):
+  output_size: List[int] = dataclasses.field(
+      default_factory=list)
+  stuff_classes_offset: int = 0
```
Comment (arashwan) [inline official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py:80]: I feel this offset is important even in the absence of the panoptic generator, right? can we have it as part of task/model.

File: official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py (hunk 2 above)
Comment (arashwan) [reply official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py:97]: remove \

File: official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py [hunk 7]
```diff
@@ -94,6 +110,16 @@ class Losses(maskrcnn.Losses):
   semantic_segmentation_weight: float = 1.0
 
 
+@dataclasses.dataclass
+class PanopticQualityEvaluator(hyperparams.Config):
+  """Panoptic Quality Evaluator config."""
+  num_categories: int = 2
+  ignored_label: int = 0
```
Comment (arashwan) [inline official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py:117]: ignore_label (no 'd').

File: official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py [hunk 8]
```diff
@@ -115,7 +141,8 @@ class PanopticMaskRCNNTask(maskrcnn.MaskRCNNTask):
   # 'all': Initialize all modules
   init_checkpoint_modules: Optional[List[str]] = dataclasses.field(
       default_factory=list)
-
+  evaluate_panoptic_quality: bool = True
```
Comment (arashwan) [inline official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py:144]: We already have generate_panoptic_masks, can you remove this one.

File: official/vision/beta/projects/panoptic_maskrcnn/tasks/__init__.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Tasks package definition."""
+
+from official.vision.beta.projects.panoptic_maskrcnn.tasks import panoptic_maskrcnn
```
Comment (arashwan) [inline official/vision/beta/projects/panoptic_maskrcnn/tasks/__init__.py:18]: remove.

File: official/vision/beta/projects/panoptic_maskrcnn/tasks/panoptic_maskrcnn.py [hunk 10]
```diff
@@ -207,6 +209,8 @@ def build_metrics(self, training: bool = True) -> List[
       tf.keras.metrics.Metric]:
     """Build detection metrics."""
     metrics = []
+    num_segmentation_classes = \
```
Comment (arashwan) [inline official/vision/beta/projects/panoptic_maskrcnn/tasks/panoptic_maskrcnn.py:212]: don't use \

File: official/vision/beta/projects/panoptic_maskrcnn/tasks/panoptic_maskrcnn.py [hunk 11]
```diff
@@ -239,9 +243,24 @@ def build_metrics(self, training: bool = True) -> List[
...
           rescale_predictions=rescale_predictions,
           dtype=tf.float32)
+      
+      if self.task_config.evaluate_panoptic_quality:
+        assert (
+            self.task_config.validation_data.parser.include_panoptic_masks,
+            '`include_panoptic_masks` should be set to True when computing '
+            'panoptic quality')
+        pq_config = self.task_config.panoptic_quality_evaluator
+        self.panoptic_quality_metric = \
```
Comment (arashwan) [inline official/vision/beta/projects/panoptic_maskrcnn/tasks/panoptic_maskrcnn.py:256]: same, don't use \.

File: official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py (hunk 7 above)
Comment (srihari-humbarwadi) [reply official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py:117]: I was trying to match it with the PQ evaluator args. Should I change it here as well?
https://github.com/tensorflow/models/blob/2af8ebcd93c5449164b8f1b4d7ee2e15dd068040/official/vision/beta/evaluation/panoptic_quality_evaluator.py#L34-L40

**Review by arashwan — APPROVED** [review]

**Review by jaeyounkim — APPROVED** [review]
